src/services/likes.service.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  createOne,
+  getOne,
+  deleteOne,
+  updateOne,
+} from "../repositories/likes.repository";
+import { getUserById } from "../repositories/user.repository";
+import { getTweetById, incrementTweetStats } from "./tweets.service";
+
+export const changeLike = async (tweetId: string, userId: string) => {
+  if (getUserById(userId) && getTweetById(tweetId)) {
+    const exists = !!(await getOne({ userId, tweetId }));
+    if (!exists) {
+      await incrementTweetStats(tweetId, "stats.likes", 1);
+      return createOne(tweetId, userId);
+    } else {
+      await incrementTweetStats(tweetId, "stats.likes", -1);
+      return deleteOne({ tweetId, userId });
```

Comment:
I'd recommend make it a bit smarter, to reduce count of requessts:
1. try to delete like with userId and tweetId
2. if it failed, it means, nothing to delete, then create a new one and increase stats
3. if it success, decrease stats

and you potentially save time of one request per each operation